packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -49,14 +44,10 @@ describe('ReactDOMInput', () => {
     var node = ReactDOM.findDOMNode(stub);
     expectDev(console.error.calls.count()).toBe(1);
 
-    // Simulate a native change event
-    setUntrackedValue(node, 'giraffe');
-
     // This must use the native event dispatching. If we simulate, we will
     // bypass the lazy event attachment system so we won't actually test this.
-    var nativeEvent = document.createEvent('Event');
-    nativeEvent.initEvent('change', true, true);
-    node.dispatchEvent(nativeEvent);
+    setUntrackedValue.call(node, 'giraffe');
+    ReactTestUtils.SimulateNative.change(node);
```

Comment:
Hm. What happens if you use the native event dispatching like it was before? Does the test still pass? If so, we should ideally use that mechanism instead since it doesn't rely on by-passing the document event listeners. 

`ReactTestUtils.SimulateNative` is reaching into internals in a way. That's exactly what we'd like to avoid with these changes.

Reply:
Totally agree, I'm gonna check it out right now and post here the results.

Reply:
@sebmarkbage Yes, the tests are passing.

Plus, I think using something like the `node.dispatchEvent(nativeEvent);` seems to be a very flexible way to obtain the expected results without reaching those internals.

Reply:
Btw I verified that using `SimulateNative` would have still passed the test if I removed [this line](https://github.com/facebook/react/blob/5f6159fa84a5a5abc8177638bf652288312113be/packages/react-dom/src/client/ReactDOMFiberComponent.js#L535-L537). Whereas with native dispatch it is covered.